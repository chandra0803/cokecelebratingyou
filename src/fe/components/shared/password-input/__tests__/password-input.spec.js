import React from 'react';
import PasswordInput from '../password-input.js';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';

const mockOptions = {
    label: 'Password',
    name: 'password',
    tabIndex: '1',
    passwordVisible: false,
    disabled: false
};

const helpComponent = <a> I forgot my password </a>;

test( 'Password input renders depending on label passed in', () => {
    const component = renderer.create(
        <PasswordInput
            { ...mockOptions }
            onChange={ () => true }
            onKeyPress={ () => true }
            helpComponent= { helpComponent }/>
    );

    const tree = component.toJSON();
    expect( tree ).toMatchSnapshot();
} );

it ( 'executes togglePasswordVisible prop when the user clicks on eye icon ', () => {
    const props = {
        ...mockOptions
    };

    const tree = mount( <PasswordInput { ...props }/> );

    tree.setState( { passwordVisible: true } );

    const input = tree.find( '.eye' );

    input.simulate( 'click' );

    expect( tree.state().passwordVisible ).toBe( false );
} );
