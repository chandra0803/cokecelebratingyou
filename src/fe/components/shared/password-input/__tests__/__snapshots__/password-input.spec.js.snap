// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Password input renders depending on label passed in 1`] = `
<div
  className="password-input"
>
  <div
    className="input-wrap"
  >
    <a>
       I forgot my password 
    </a>
    <label
      className="control-label"
      htmlFor="password1"
    >
      Password
    </label>
    <span
      className="eye"
      onClick={[Function]}
    >
      <svg
        className="icon"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,3.9C4.2,3.9,0,9.6,0,12s4.2,8.1,12,8.1s12-5.8,12-8.1S19.8,3.9,12,3.9z M12,18.7c-7.2,0-10.6-5.4-10.6-6.7 S4.8,5.3,12,5.3s10.6,5.4,10.6,6.7S19.2,18.7,12,18.7z"
        />
        <path
          d="M12,7c-2.8,0-5,2.2-5,5s2.2,5,5,5s5-2.2,5-5S14.8,7,12,7z M12,15.6c-2,0-3.6-1.6-3.6-3.6c0-2,1.6-3.6,3.6-3.6 c2,0,3.6,1.6,3.6,3.6C15.6,14,14,15.6,12,15.6z"
        />
      </svg>
    </span>
    <input
      autoFocus={undefined}
      className="gq-input"
      disabled={false}
      name="password"
      onBlur={undefined}
      onChange={[Function]}
      onFocus={undefined}
      onKeyPress={[Function]}
      tabIndex="1"
      type="text"
      value={undefined}
    />
  </div>
</div>
`;
